### PR TITLE
feat: kmp demo

### PR DESCRIPTION
**Pull request by buenaflor**
Add additional KMP demo that can be used for the kotlinconf

includes:
 - Koin dependency injection
 - ViewModels

### CODE REVIEW

File: sentry-samples/kmp-app-mvvm-di/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+## Shared Features
+ - Dependency Injection with Koin
+ - ViewModels
+ - Sentry Setup
+
+## Getting Started
+
+### DSN
+If you need to change the `DSN` you can do so in the `SentrySetup.kt` file in the `shared` module.
+
```
Comment (marandaneto) [inline sentry-samples/kmp-app-mvvm-di/README.md:14]: Missing `pod install` on `sentry-samples/kmp-app/iosApp`.

File: sentry-samples/kmp-app/iosApp/iosApp.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -165,6 +166,25 @@
...
+				"${DWARF_DSYM_FOLDER_PATH}/${DWARF_DSYM_FILE_NAME}/Contents/Resources/DWARF/${TARGET_NAME}",
+			);
+			name = "Senry dsyms upload";
+			outputFileListPaths = (
+			);
+			outputPaths = (
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "if which sentry-cli >/dev/null; then\nexport SENTRY_ORG=sentry-sdks\nexport SENTRY_PROJECT=sentry-kotlin-multiplatform\nexport SENTRY_AUTH_TOKEN=14e7bf400bc449589fdc32fad8beaad003456b05b784403fb11d109b6276cefe\nERROR=$(sentry-cli upload-dif \"$DWARF_DSYM_FOLDER_PATH\" 2>&1 >/dev/null)\nif [ ! $? -eq 0 ]; then\necho \"warning: sentry-cli - $ERROR\"\nfi\nelse\necho \"warning: sentry-cli not installed, download from https://github.com/getsentry/sentry-cli/releases\"\nfi\n";
```
Comment (marandaneto) [inline sentry-samples/kmp-app/iosApp/iosApp.xcodeproj/project.pbxproj:186]: Please rotate the `SENTRY_AUTH_TOKEN` token and remove it from here, just add a place holder.

Comment (buenaflor) [reply sentry-samples/kmp-app/iosApp/iosApp.xcodeproj/project.pbxproj:186]: weird that it shows up, I've specifically removed it. but yeah that shouldn't be there at all

File: sentry-samples/kmp-app-mvvm-di/README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# Sentry KMP Demo App
+
+This is a demo app for the Sentry Kotlin Multiplatform SDK that includes a native iOS app and a native Android app with shared code.
+
+## Shared Features
+ - Dependency Injection with Koin
+ - ViewModels
+ - Sentry Setup
+
+## Getting Started
```
Comment (marandaneto) [inline sentry-samples/kmp-app-mvvm-di/README.md:10]: Missing note about Cocoapods and `export LANG=en_US.UTF-8`

Comment (marandaneto) [reply sentry-samples/kmp-app-mvvm-di/README.md:10]: Getting error:

> * Exception is:
> org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':sentry-kotlin-multiplatform:podBuildSentryAppletvos'.
> 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:142)
> 	at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:282)
> 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:140)
> 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:128)
> 	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:77)
> 	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
> 	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
> 	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
> 	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
> 	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
> 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
> 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
> 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
> 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
> 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
> 	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
> 	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
> 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
> 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
> 	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
> 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
> 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
> 	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:69)
> 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:327)
> 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:314)
> 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:307)
> 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:293)
> 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:417)
> 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:339)
> 	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
> 	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
> Caused by: java.lang.IllegalStateException: Executing of 'xcodebuild -project Pods.xcodeproj -scheme Sentry -sdk appletvos -configuration Release' failed with code 70 and message: 
> 
> Command line invocation:
>     /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -project Pods.xcodeproj -scheme Sentry -sdk appletvos -configuration Release
> 
> User defaults from command line:
>     IDEPackageSupportUseBuiltinSCM = YES
> 
> Build settings from command line:
>     SDKROOT = appletvos16.4

@buenaflor ideas?

File: sentry-samples/kmp-app-mvvm-di/README.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# Sentry KMP Demo App
```
Comment (marandaneto) [inline sentry-samples/kmp-app-mvvm-di/README.md:1]: @buenaflor what are the steps for compiling and running on Android and iOS? Which IDE? I can't run on Xcode nor AS right now.

Comment (marandaneto) [reply sentry-samples/kmp-app-mvvm-di/README.md:1]: >  SDK location not found. Define a valid SDK location with an ANDROID_HOME environment variable or by setting the sdk.dir path in your project's local properties file at '/Users/marandaneto/Github/sentry-kotlin-multiplatform/local.properties'.

Mention to export `ANDROID_HOME`

Comment (marandaneto) [reply sentry-samples/kmp-app-mvvm-di/README.md:1]: Mention `:sentry-samples:kmp-app:androidApp:assembleDebug` would compile the Android version

File: sentry-samples/kmp-app-mvvm-di/androidApp/build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,53 @@
+plugins {
```
[the remark this answers is not in the document]
Comment (marandaneto) [reply sentry-samples/kmp-app-mvvm-di/androidApp/build.gradle.kts:1]: After disabling coreLibraryDesugaring:

> e: file:///Users/x/Github/sentry-kotlin-multiplatform/sentry-samples/kmp-app-mvvm-di/shared/src/androidMain/kotlin/io/sentry/kmp/demo/models/ViewModel.kt:5:27 Unresolved reference: viewModelScope
> e: file:///Users/x/Github/sentry-kotlin-multiplatform/sentry-samples/kmp-app-mvvm-di/shared/src/androidMain/kotlin/io/sentry/kmp/demo/models/ViewModel.kt:8:49 Unresolved reference: androidXViewModelScope
>

File: sentry-samples/kmp-app-mvvm-di/README.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+You can use Android Studio to run both the Android and iOS sample apps.
+
+The android target is available as `sentry-samples.kmp-app-mvvm-di.androidApp` automatically.
+The iOS target needs to be configured: add a new run configuration and select an iOS application as a new target.
+You can then select the `iosApp.xcworkspace` file for the required `Xcode project file`.
+
+### Android
+- Export your `ANDROID_HOME` environment variable if you haven't done already.
+- You can run `./gradlew :sentry-samples:kmp-app:androidApp:assembleDebug` to compile the Android app.
```
Comment (marandaneto) [inline sentry-samples/kmp-app-mvvm-di/README.md:24]:
```suggestion
- You can run `./gradlew :sentry-samples:kmp-app:androidApp:assembleDebug` to compile the Android app.
- You can run `xcodebuild -workspace ./sentry-samples/kmp-app-mvvm-di/iosApp/iosApp.xcworkspace -scheme iosApp -configuration Debug -sdk iphonesimulator -arch arm64` to compile the iOS app.
```

File: sentry-samples/kmp-app-mvvm-di/README.md (hunk 3 above)
Comment (marandaneto) [reply sentry-samples/kmp-app-mvvm-di/README.md:10]: Missing to download the apple tv + watch os xcode files, only had for mac and ios

**Comment by marandaneto** [comment]
@buenaflor I face the same errors as CI did https://github.com/getsentry/sentry-kotlin-multiplatform/actions/runs/4667538687/jobs/8263458621?pr=71
do `make compile`

**Comment by marandaneto** [comment]
@buenaflor So at the end of the day this didn't work at all for iOS, I tried myself but could not figure it out quickly.
Would you mind taking a stab and making it green on CI? I probably face the same errors as CI.

**Comment by buenaflor** [comment]
Yeah I'll take a look at it

**Comment by buenaflor** [comment]
added in #82

**Pull request closed without merging**